docs/source/api.rst

```diff
@@ -700,3 +703,310 @@ retworkx API
         :param int edge: The index of the edge to remove
 
 .. _petgraph: https://github.com/bluss/petgraph
+
+.. py:class:: PyDiGraph
+   A class for creating directed graph
+
+   The PyDiGraph class is constructed using the Rust library `petgraph`_ around
+   the ``StableGraph`` type. The limitations and quirks with this library and
+   type dictate how this operates. The biggest thing to be aware of when using
+   the PyDiGraph class is that an integer node and edge index is used for accessing
+   elements on the graph, not the data/weight of nodes and edges.
+
+   If you would like to use a PyDiGraph object as a directed acyclic graph you
+   can do this by enabling realtime cycle checking. For example::
+
+       import retworkx
+       dag = retworkx.PyDiGraph()
+       dag.check_cycle = True
+
+   or at object creation::
+
+       import retworkx
+       dag = retworkx.PyDiGraph(True)
```

Comment:
weird to see a flag with no kwarg / more readable as `check_cycle=True`

Reply:
Sure, done in: https://github.com/Qiskit/retworkx/pull/88/commits/34875a73502eb4adee6b2ab46e1db9f3aadb5152